packages/web-platform/web-rsbuild-plugin/src/plugins/prefetch-worker.ts

```
@@ -0,0 +1,129 @@
+// Copyright 2024 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+
+import type { HtmlRspackPlugin } from '@rsbuild/core';
+import path from 'path';
+import type { Compiler, Compilation } from '@rspack/core';
+
+export class PrefetchWorkerPlugin {
+  resourceHints: HtmlRspackPlugin.HtmlTagObject[] = [];
+  chunkIds: Set<string> = new Set();
+
+  HtmlPlugin: HtmlRspackPlugin;
+
+  constructor(
+    options: { HtmlPlugin: HtmlRspackPlugin },
+  ) {
+    this.HtmlPlugin = options.HtmlPlugin;
+  }
+
+  apply(compiler: Compiler) {
+    if (
+      compiler.options.mode !== 'production'
+      || typeof compiler.options.output?.publicPath === 'function'
+    ) {
+      console.warn(
```

I would prefer throwing an error instead of console a warning and bailout.